src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -239,7 +239,8 @@ public Task<TResponse> GetResponseAsync()
 
             if (httpResponse.StatusCode != HttpStatusCode.OK)
             {
-                return new Status(StatusCode.Cancelled, "Bad gRPC response. Expected HTTP status code 200. Got status code: " + (int)httpResponse.StatusCode);
+                var statusCode = MapHttpStatusToGrpcCode(httpResponse.StatusCode);
```

Comment:
This looks like it may be too early to perform this translation. The C code checks for a grpc-status value before mapping the HTTP2 status code, regardless of what the HTTP2 status code is--if I'm reading it correctly. [Java also looks for grpc-status first](https://github.com/grpc/grpc-java/blob/81da3eb95be37fa0647ce8da2e19de96ab84c600/core/src/main/java/io/grpc/internal/Http2ClientStreamTransportState.java#L181-L201). Not sure about Go...